app/src/androidTest/java/org/mozilla/tv/firefox/ui/BasicNavigationTest.kt

```diff
@@ -0,0 +1,146 @@
+
+        var preReloadTitle: String? = null
+
+        // Navigate to several URLs with the URL bar.
+        navigationOverlay {
+            assertCanGoBackForward(false, false)
+            assertCanReload(false)
+            assertURLBarDisplaysHint()
+
+        }.enterUrlAndEnterToBrowser(pageUrls[0]) {
+            assertTestContent(BODY_TEXT[0])
+        }.openOverlay {
+            assertCanGoBackForward(true, false) // TODO: Unexpected. Fix in #1347.
+            assertCanReload(true)
+            assertURLBarText(pageUrlStrs[0])
+
+        }.enterUrlAndEnterToBrowser(pageUrls[1]) {
+            assertTestContent(BODY_TEXT[1])
+        }.openOverlay {
+            assertCanGoBackForward(true, false)
+            assertCanReload(true)
+            assertURLBarText(pageUrlStrs[1])
+
+        }.enterUrlAndEnterToBrowser(pageUrls[2]) {
+            assertTestContent(BODY_TEXT[2])
+        }.openOverlay {
+            assertCanGoBackForward(true, false)
+            assertCanReload(true)
+            assertURLBarText(pageUrlStrs[2])
+
+            // Verify back behavior (to the beginning of the stack).
+            goBack()
+            assertCanGoBackForward(true, true)
+            assertCanReload(true)
+            assertURLBarText(pageUrlStrs[1])
+        }.closeToBrowser {
+            assertTestContent(BODY_TEXT[1])
+
+        }.openOverlay {
+            goBack()
+            assertCanGoBackForward(true, true) // TODO: unexpected. Fix in #1347.
+            assertCanReload(true)
+            assertURLBarText(pageUrlStrs[0])
+        }.closeToBrowser {
+            assertTestContent(BODY_TEXT[0])
+
+        }.openOverlay {
+            // Verify forward behavior (to the end of the stack).
+            goForward()
+            assertCanGoBackForward(true, true)
+            assertCanReload(true)
+            assertURLBarText(pageUrlStrs[1])
+        }.closeToBrowser {
+            assertTestContent(BODY_TEXT[1])
+
+        }.openOverlay {
+            goForward()
+            assertCanGoBackForward(true, false)
+            assertCanReload(true)
+            assertURLBarText(pageUrlStrs[2])
+        }.closeToBrowser {
+            assertTestContent(BODY_TEXT[2])
+            preReloadTitle = getPageTitle()
+
+        }.openOverlay {
+            // Verify reload behavior.
+            reload()
+            assertCanGoBackForward(true, false)
+            assertCanReload(true)
+            assertURLBarText(pageUrlStrs[2])
+        }.closeToBrowser {
+            assertTestContent(BODY_TEXT[2]) // different on each reload. Use date? Verify date different?
+
+            // We want to ensure the page is actually reloaded after reload is pressed. This
+            // particular page's title changes each time the page is reloaded so we verify the
+            // current title is different.
+            assertNotEquals(preReloadTitle, getPageTitle())
+        }
+    }
+    /* ktlint-enable no-blank-line-before-rbrace */
```

Comment:
This test can also replace the former PageLoadTest.java (feel free to delete it).

Reply:
I wanted to leave in a single test in that handles network page load so I left PageLoadTest around. However, it would be a good idea to make it clear that's why it still exists (@Baron-Severin can you make this part of the non-critical UI test follow-up? Or maybe we should just fix that now...).